app/api/streams/StreamApplyDelta.js

```diff
@@ -53,45 +51,47 @@ module.exports = async ( req, res ) => {
     console.log(delta.revision_A)
 
     //checks if delta can be applied. We have to make sure that the streamId of original stream should be the same as revision_A id.
-    if ( delta.revision_A != stream.streamId ) {
-      console.log("Meeep - mismatched versions")
+    if ( delta.revision_A.id != stream.streamId ) {
       throw new Error("Mismatched revision/streamId.")
     } else {
   
-      if( delta.created ) {
-
-        let common = stream.objects.filter( obj => delta.created.indexOf(obj._id.toString()) !== -1 )
-
-        if(common.length != 0)
-        {
-          // TODO: Finish BAD DELTA check
-         // Check objects in stream. If there exist objects that have the same id of the `delta.created`, the delta is bad.
-
+      // objects do be added to the stream
+      let objsToAdd = delta.delta.created
+      for (i = 0; i < objsToAdd.length; i++) {      
+        let objToAdd = objsToAdd[i]
+        // adds object to the stream's objects field
+        if (stream.objects.indexOf(objToAdd._id) === -1) { // check if object does NOT exist in the stream
+          console.log("This object needs to be created.")
+          stream.objects.push(objToAdd)
+          await stream.save()
+        } else {
+          // BAD DELTA CHECK
           console.log("Bad delta")
-          throw new Error("Bad delta")
+          throw new Error("Bad delta. Some delta.created objects are existing in the original stream")
         }
-
-        console.log( delta.created )
-        let objs = await BulkObjectSave(delta.created, req.user)
-        stream.objects = new Set( [...stream.objects, ...objs.map( o => o._id.toString() ) ])
       }
 
       // objects to be deleted from the stream
-      if(delta.deleted){
-        // TODO: Finish BAD DELTA check
-        // Check objects in stream. If any of the `delta.deleted` objects are not in the stream, the delta is bad.
-        let objsToDelete = delta.deleted
-        stream.objects = stream.objects.filter( obj => delta.deleted.indexOf(obj._id.toString()) !== -1 )
+      let objsToDelete = delta.delta.deleted
+      for (i = 0; i < objsToDelete.length; i++) {
+        let objToDelete = objsToDelete[i]
+        // removes object from the stream's objects field
+        if (stream.objects.indexOf(objToDelete._id) !== -1) { // check if object exists in the stream
+          console.log("This object needs to be removed.")
+          removeAllElements(stream.objects, objToDelete._id)
+          await stream.save()
+        } else {
+          // BAD DELTA CHECK
+          console.log("Bad delta")
+          throw new Error("Bad delta. Some delta.deleted objects are not exsiting in the original stream.")
+        }
```

Comment:
I reimplemented what I originally wrote for this StramApplyDelta endpoint. I know @didimitrie you tried to be more synthetic with arrows functions, but I could not make sense of it. I tested your code against with postman and it did not work properly. Tried to fix it first but I just ended up going back to the original version. Now this works for sure as I tested against dummy data and the delta applies well, either for removing or adding objects. I still tried to take as much as I could what you wrote into account, such as the bad delta checks which are now there. Other than that, I think it's ready for merging @alelom (besides the failed appveyor checks).